ecosystem/gstreamer_plugin/gst_mtl_st40p_tx.c

```diff
@@ -0,0 +1,426 @@
+
+#define gst_mtl_st40p_tx_parent_class parent_class
+G_DEFINE_TYPE_WITH_CODE(Gst_Mtl_St40p_Tx, gst_mtl_st40p_tx, GST_TYPE_BASE_SINK,
+                        GST_DEBUG_CATEGORY_INIT(gst_mtl_st40p_tx_debug, "mtl_st40p_tx", 0,
+                                                "MTL St2110 st40 transmission sink"));
+
+GST_ELEMENT_REGISTER_DEFINE(mtl_st40p_tx, "mtl_st40p_tx", GST_RANK_NONE,
+                            GST_TYPE_MTL_ST40P_TX);
+
+static void gst_mtl_st40p_tx_set_property(GObject* object, guint prop_id,
+                                          const GValue* value, GParamSpec* pspec);
+static void gst_mtl_st40p_tx_get_property(GObject* object, guint prop_id, GValue* value,
+                                          GParamSpec* pspec);
+static void gst_mtl_st40p_tx_finalize(GObject* object);
+
+static gboolean gst_mtl_st40p_tx_sink_event(GstPad* pad, GstObject* parent,
+                                            GstEvent* event);
+static GstFlowReturn gst_mtl_st40p_tx_chain(GstPad* pad, GstObject* parent,
+                                            GstBuffer* buf);
+
+static gboolean gst_mtl_st40p_tx_start(GstBaseSink* bsink);
+static gboolean gst_mtl_st40p_tx_session_create(Gst_Mtl_St40p_Tx* sink);
+
+static void gst_mtl_st40p_tx_class_init(Gst_Mtl_St40p_TxClass* klass) {
+  GObjectClass* gobject_class;
+  GstElementClass* gstelement_class;
+  GstBaseSinkClass* gstbasesink_class;
+
+  gobject_class = G_OBJECT_CLASS(klass);
+  gstelement_class = GST_ELEMENT_CLASS(klass);
+  gstbasesink_class = GST_BASE_SINK_CLASS(klass);
+
+  gst_element_class_set_metadata(
+      gstelement_class, "MtlTxSt40Sink", "Sink/Metadata",
+      "MTL transmission plugin for SMPTE ST 2110-40 standard (ancillary data)",
+      "Marek Kasiewicz <marek.kasiewicz@intel.com>");
+
+  gst_element_class_add_static_pad_template(gstelement_class,
+                                            &gst_mtl_st40p_tx_sink_pad_template);
+
+  gobject_class->set_property = GST_DEBUG_FUNCPTR(gst_mtl_st40p_tx_set_property);
+  gobject_class->get_property = GST_DEBUG_FUNCPTR(gst_mtl_st40p_tx_get_property);
+  gobject_class->finalize = GST_DEBUG_FUNCPTR(gst_mtl_st40p_tx_finalize);
+  gstbasesink_class->start = GST_DEBUG_FUNCPTR(gst_mtl_st40p_tx_start);
+
+  gst_mtl_common_init_general_argumetns(gobject_class);
+}
+
+static gboolean gst_mtl_st40p_tx_start(GstBaseSink* bsink) {
+  struct mtl_init_params mtl_init_params = {0};
+
+  Gst_Mtl_St40p_Tx* sink = GST_MTL_ST40P_TX(bsink);
+
+  GST_DEBUG_OBJECT(sink, "start");
+  GST_DEBUG("Media Transport Initialization start");
+  gst_base_sink_set_async_enabled(bsink, FALSE);
+
+  sink->mtl_lib_handle =
+      gst_mtl_common_init_handle(&mtl_init_params, &(sink->devArgs), &(sink->log_level));
+
+  if (!sink->mtl_lib_handle) {
+    GST_ERROR("Could not initialize MTL");
+    return FALSE;
+  }
+  sink->retry_frame = 10;
+
+  gst_mtl_st40p_tx_session_create(sink);
+
+  gst_element_set_state(GST_ELEMENT(sink), GST_STATE_PLAYING);
+
```

Comment:
are you installing your parameters ? 

i can see PROP_ST40P_TX_RETRY but no 
  g_object_class_install_property(
      gobject_class, PROP_ST40P_TX_RETRY
      
      I may be blind though

Reply:
nope, you're not blind. I was. I'm already fixing this

Reply:
done